src/services/jsTyping.ts

```diff
@@ -26,6 +26,24 @@ namespace ts.JsTyping {
         typings?: string;
     }
 
+    export interface CachedTyping {
+        typingLocation: string;
+        timestamp: number;
+    }
+
+    export function isTypingExpired(typing: JsTyping.CachedTyping | undefined) {
+        const comparisonDate = new Date();
+        const currentMonth = comparisonDate.getMonth();
+        if (currentMonth) {
+            comparisonDate.setMonth(11);
+            comparisonDate.setFullYear(comparisonDate.getFullYear() - 1);
+        }
+        else {
+            comparisonDate.setMonth(currentMonth - 1);
+        }
+        return !typing || typing.timestamp < comparisonDate.getTime();

```

Comment:
Easier to just subtract 1000 * 60 * 60 * 24 * 30 or whatever other expression makes sense

Reply:
Cleaned up.